xdg-terminal-exec

```
@@ -175,55 +176,49 @@ save_cache() {
+				debug "directive	$entry_id ${XTE_CACHE_ENABLED+'(ignored)'}"
 				CACHE_ENABLED=${XTE_CACHE_ENABLED-true}
 				;;
-			/disable_cache)
-				debug "found '$line' directive${XTE_CACHE_ENABLED+ (ignored)}"
+			'/disable_cache:')
+				debug "directive	$entry_id ${XTE_CACHE_ENABLED+'(ignored)'}"
 				CACHE_ENABLED=${XTE_CACHE_ENABLED-false}
 				;;
-
-			# `[The extensionless entry filename] should be a valid D-Bus well-known name.`
-			# `a sequence of non-empty elements separated by dots (U+002E FULL STOP),
-			# none of which starts with a digit, and each of which contains only characters from the set [a-zA-Z0-9-_]`
-			# Stricter parts seem to be related only to reversed DNS notation but not common naming
-			# i.e. there is `2048-qt.desktop`.
-			# I do not know of any terminal that starts with a number, but it's valid.
-
-			# Catch and validate potential entry ID with action ID (be graceful about an empty one)
-			[a-zA-Z0-9_]*)
-				# consider only the first ':' as a delimiter
-				IFS=':' read -r entry_id action_id <<- EOL
-					$line
-				EOL
-				if validate_entry_id "${entry_id}" && validate_action_id "${action_id}"; then
-					ENTRY_IDS=${ENTRY_IDS:+${ENTRY_IDS}${N}}$line
-					debug "added entry ID with action ID '$line'"
-				else
-					error "Discarded possibly misspelled entry '$line'"
-				fi
+			# Ignore comments and empty lines
+			'#'* | ':') ;;
+			# Sorry for the liberal paraphrasing
+			# `[The extensionless entry filename comprises of dot separated elements] each of which contains only [...] [A-Za-z0-9-_]`
+			# Catch and complain on invalid IDs
+			*[![:alnum:]_.-]*'.desktop:'* | *'.desktop:'*[![:alnum:]_.-]* | '.desktop:'*)
+				error "$config_path: line $linenum: discarded invalid entry"
+				;;
+			# Catch valid entry ID
+			[[:alnum:]_]*'.desktop:'*)
+				READ_ENTRY_IDS=${READ_ENTRY_IDS:+${READ_ENTRY_IDS}${N}}$entry_id:$action_id
+				debug "$entry_id	: $action_id"
+				;;
+			# Catch and complain on invalid lines
+			*)
```

this is too aggressive, we need to silently drop unrecognized stuff, only react on invalid entries (things containing `.desktop`, but otherwise invalid).

---

Would you rather it be a debug message, or nothing at all?

---

I liked `validate_entry_id()`, `validate_action_id()`, we could use them if line contains `.desktop` and produce error on negative result. And just do not bother with anything not explicitly recognized and not containing `.desktop`.

---

...not an error, a warning.

---

I like this more because all the logic is in one place
Simpler to write tests for too

I changed the catch all to a debug message, is this good?

---

On the topic, as the file format is part of the proposed spec, I think the reference implementation should be responsible for validating the files. At least to some degree